packages/wdio-browserstack-service/src/launcher.ts

```diff
@@ -192,6 +194,10 @@ export default class BrowserstackLauncherService implements Services.ServiceInst
         } catch (err: unknown) {
             PercyLogger.error(`Error while setting best platform for Percy snapshot at worker start ${err}`)
         }
+
+        if (!isBrowserstackInfra(this._config)){
+            caps = await AiHandler.setup(this._config, this.browserStackConfig, this._options, caps)
```

Comment:
What should be the behaviour in case this promise gets rejected? Consider adding try catch.

Reply:
The entire logic of setup function is inside try catch. Haven't added it here to keep the code clean.

Reply:
Let's add try/catch here.